cmd/root.go

```diff
@@ -43,71 +44,28 @@ const (
 	scorecardShort = "Security Scorecards"
 )
 
-var rootCmd = &cobra.Command{
-	Use:   scorecardUse,
-	Short: scorecardShort,
-	Long:  scorecardLong,
-	Run:   scorecardCmd,
-}
-
-var opts = options.New()
-
-//nolint:gochecknoinits
-func init() {
-	rootCmd.Flags().StringVar(&opts.Repo, "repo", "", "repository to check")
-	rootCmd.Flags().StringVar(&opts.Local, "local", "", "local folder to check")
-	rootCmd.Flags().StringVar(&opts.Commit, "commit", options.DefaultCommit, "commit to analyze")
-	rootCmd.Flags().StringVar(
-		&opts.LogLevel,
-		"verbosity",
-		options.DefaultLogLevel,
-		"set the log level",
-	)
-	rootCmd.Flags().StringVar(
-		&opts.NPM, "npm", "",
-		"npm package to check, given that the npm package has a GitHub repository")
-	rootCmd.Flags().StringVar(
-		&opts.PyPI, "pypi", "",
-		"pypi package to check, given that the pypi package has a GitHub repository")
-	rootCmd.Flags().StringVar(
-		&opts.RubyGems, "rubygems", "",
-		"rubygems package to check, given that the rubygems package has a GitHub repository")
-	rootCmd.Flags().StringSliceVar(
-		&opts.Metadata, "metadata", []string{}, "metadata for the project. It can be multiple separated by commas")
-	rootCmd.Flags().BoolVar(&opts.ShowDetails, "show-details", false, "show extra details about each check")
-	checkNames := []string{}
-	for checkName := range checks.GetAll() {
-		checkNames = append(checkNames, checkName)
-	}
-	rootCmd.Flags().StringSliceVar(&opts.ChecksToRun, "checks", []string{},
-		fmt.Sprintf("Checks to run. Possible values are: %s", strings.Join(checkNames, ",")))
-
-	// TODO(cmd): Extract logic
-	if options.IsSarifEnabled() {
-		rootCmd.Flags().StringVar(&opts.PolicyFile, "policy", "", "policy to enforce")
-		rootCmd.Flags().StringVar(&opts.Format, "format", options.FormatDefault,
-			"output format allowed values are [default, sarif, json]")
-	} else {
-		rootCmd.Flags().StringVar(&opts.Format, "format", options.FormatDefault,
-			"output format allowed values are [default, json]")
+// New creates a new instance of the scorecard command.
+func New() *cobra.Command {
+	cmd := &cobra.Command{
```

Comment:
@n3wscott -- Could you say more about the utility of calling options twice?

I kept it scoped to the package instead of the function because the `serveCmd()` command expects the root options to be available.

Reply:
One goal of moving these things is to make them testable, but if your test calls new in parallel you will have a race on the state of the global options.

Reply:
Okay, cool. I'll keep it as-is for now, but maybe revisit that as we tease some more of the structure apart.
Appreciate the review, Scott!